includes/steps/class-step-update-user.php

```diff
@@ -0,0 +1,687 @@
+
+		if ( rgobj( $field, 'multipleFiles' ) ) {
+			$is_applicable_field = false;
+		}
+
+		return $is_applicable_field;
+	}
+
+	/**
+	 * Prepare value map.
+	 *
+	 * @since 2.5
+	 *
+	 * @return array
+	 */
+	public function value_mappings() {
+
+		$fields = gravity_flow()->get_field_map_choices( $this->get_form_id() );
+
+		return $fields;
+	}
+
+	/**
+	 * Sets the user properties.
+	 *
+	 * @since 2.5
+	 *
+	 * @param WP_User $user
+	 */
+	public function set_user_properties( $user ) {
+
+		$entry = $this->get_entry();
+
+		$standard_meta = array(
+			'first_name',
+			'last_name',
+			'nickname',
+		);
+
+		$form = $this->get_form();
+
+		$dirty = false;
+
+		foreach ( $standard_meta as $meta_key ) {
+			$this->log_debug( sprintf( 'Processing meta item: %s', $meta_key ) );
+
+			if ( ! $this->{$meta_key} ) {
+				$this->log_debug( 'Meta item is empty. Skipping it.' );
+				continue;
+			}
+
+			$this->{$meta_key};
+			$value = gravity_flow()->get_field_value( $form, $entry, $this->{$meta_key} );
+			$this->log_debug( sprintf( 'Meta item mapped to field: %s; Value: %s', $meta_key, $value ) );
+
+			$result = update_user_meta( $user->ID, $meta_key, $value );
+			$this->log_debug( sprintf( 'Result: %s', var_export( (bool) $result, 1 ) ) );
+
+			$dirty = true;
+		}
+
+		if ( $this->display_name ) {
+			$display_name = $this->get_display_name( $user );
+			$this->update_user_property( $user->ID, 'display_name', $display_name );
+			$dirty = true;
+		}
+
+		if ( $this->email ) {
+			$value = gravity_flow()->get_field_value( $form, $entry, $this->email );
+			$this->update_user_property( $user->ID, 'user_email', $value );
+			$dirty = true;
+		}
+
+		if ( is_array( $this->user_meta ) || ! empty( $this->user_meta ) ) {
+			$wc_meta = array();
+			$wc_active = is_plugin_active( 'woocommerce/woocommerce.php' );
+			foreach ( $this->user_meta as $meta_key => $meta_item ) {
+				$is_custom_value = $meta_item['value'] == 'gf_custom';
+
+				$meta_key   = $meta_item['key'] == 'gf_custom' ? $meta_item['custom_key'] : $meta_item['key'];
+				$meta_value = $is_custom_value ? $meta_item['custom_value'] : $meta_item['value'];
+
+				$this->log_debug( sprintf( 'Adding meta item: %s', $meta_key ) );
+
+				// skip empty meta items
+				if ( ! $meta_key || ! $meta_value ) {
+					$this->log_debug( 'Meta item is empty. Skipping it.' );
+					continue;
+				}
+
+				if ( $is_custom_value ) {
+					$value = GFCommon::replace_variables( $meta_value, $form, $entry, false, false, false, 'text' );
+					$value = do_shortcode( $value );
+				} else {
+					$value = gravity_flow()->get_field_value( $form, $entry, $meta_value );
+
+					if ( $wc_active && in_array( $meta_key, array(
+							'billing_state',
+							'shipping_state',
+						) ) ) {
+						$field = GFFormsModel::get_field( $form, $meta_value );
+						if ( $field->get_input_type() == 'address' && $meta_value = $field->formId . '.4' ) {
```

Comment:
Check we have a field, just in case the field is deleted and they forget to update the step.
```suggestion
						if ( $field && $field->get_input_type() == 'address' && $meta_value = $field->formId . '.4' ) {
```